.github/workflows/release.yml

```diff
@@ -0,0 +1,60 @@
+name: release
+
+on:
+  workflow_dispatch:
+    inputs:
+      is-stack-release:
+        type: boolean
+        description: Should we tag the current version as a stack release?
+        default: false
+
+permissions:
+  contents: write
+
+env:
+   GH_TOKEN: ${{ github.token }}
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-git@current
+      - uses: elastic/apm-pipeline-library/.github/actions/setup-npmrc@current
+        with:
+          secret: secret/jenkins-ci/npmjs/elasticmachine
+          vault-url: ${{ secrets.VAULT_ADDR }}
+          vault-role-id: ${{ secrets.VAULT_ROLE_ID }}
+          vault-secret-id: ${{ secrets.VAULT_SECRET_ID }}
+      - run: npm run clean
+      - run: npm ci
+      - name: Bump the beta version
+        run: |
+          echo "VERSION=$(npm version prerelease --preid=beta)" >> $GITHUB_ENV
+          echo "NPM_PACKAGE=$(jq '.name' package.json) >> $GITHUB_ENV
+      - run: git push --tags
```

Comment:
Will the `actions/checkout@v3` requires a full cloning rather than a shallow cloning when running `git push --tags`?

Reply:
Good question.

I will test this. But my assumption is that pushing won't be a problem.

Reply:
Yup. works

I tested it on my personal test repo

![image](https://user-images.githubusercontent.com/16325797/234231913-a95b5532-2e8c-4bb0-a72f-dcd5e7468999.png)

and it worked

![image](https://user-images.githubusercontent.com/16325797/234232218-ba005532-e85d-4ecc-a0e6-ee275c17c00b.png)